asserts/constraint.go

```diff
@@ -218,15 +219,23 @@ var (
 )
 
 func compileEvalAttrMatcher(cc compileContext, s string) (attrMatcher, error) {
-	if len(cc.opts.allowedOperations) == 0 {
-		return nil, fmt.Errorf("cannot compile %q constraint %q: no $OP() constraints supported", cc, s)
+	if len(cc.opts.allowedOperations) == 0 && len(cc.opts.allowedRefs) == 0 {
+		return nil, fmt.Errorf("cannot compile %q constraint %q: no $OP() or $REF constraints supported", cc, s)
 	}
+
+	if strutil.ListContains(cc.opts.allowedRefs, s[1:]) {
+		return evalAttrMatcher{op: s[1:]}, nil
+	}
+
 	ops := validEvalAttrMatcher.FindStringSubmatch(s)
 	if len(ops) == 0 || !validEvalAttrMatcherOps[ops[1]] || !strutil.ListContains(cc.opts.allowedOperations, ops[1]) {
-		oplst := make([]string, 0, len(cc.opts.allowedOperations))
+		oplst := make([]string, 0, len(cc.opts.allowedOperations)+len(cc.opts.allowedRefs))
 		for _, op := range cc.opts.allowedOperations {
 			oplst = append(oplst, fmt.Sprintf("$%s()", op))
 		}
+		for _, op := range cc.opts.allowedRefs {
+			oplst = append(oplst, fmt.Sprintf("$%s", op))
+		}
 		return nil, fmt.Errorf("cannot compile %q constraint %q: not a valid %s constraint", cc, s, strings.Join(oplst, "/"))
```

Comment:
Since the body is a bit longer now, could use a comment saying what went wrong (`FindStringSubmatch` output parsing is a bit opaque) and that this body just builds an error message to be returned.

Reply:
Added a comment, thanks